gprofiler/client.py

```diff
@@ -78,7 +79,14 @@ def _send_request(
             opts["headers"]["Content-type"] = "application/json"
             buffer = BytesIO()
             with gzip.open(buffer, mode="wt", encoding="utf-8") as gzip_file:
-                json.dump(data, gzip_file, ensure_ascii=False)  # type: ignore
+                try:
+                    json.dump(data, gzip_file, ensure_ascii=False)  # type: ignore
+                except TypeError:
+                    bad_json = str(data)
```

Comment:
I don't understand how it can happen. We are passing structured data

Reply:
As I said, it shouldn't happen in production, and only while in development. It's useful to have when adding new features.

Reply:
As commented in the other one - explain in comment that this is not expected as a real comment and it's only for devs